lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```
@@ -654,10 +675,210 @@ namespace Intl
+        if (formatted == nullptr && formattedLen == 0 && status == U_BUFFER_OVERFLOW_ERROR)
+        {
+            // when we are just counting bytes, we can ignore errors
+            AssertOrFailFast(required > 0);
+        }
+        else
+        {
+            ICU_ASSERT(status, required > 0 && required < formattedLen);
+        }
+
+        return required + 1;
+    }
+
+    // Formats `date` using the UDateFormat wrapped by `resource`. If `formatted` is non-null, the result is written into `formatted`
+    // Takes an additional parameter, fieldIterator, which is set to a IPlatformAgnosticResource* that should later be given to GetDateTimePartInfo
+    // Returns the number of characters written into `formatted` (includes a null terminator) [should always be positive]
+    int FormatDateTimeToParts(_In_ IPlatformAgnosticResource *resource, _In_ double date, _Out_writes_opt_(formattedLen) char16 *formatted,
+        _In_ int formattedLen, _Out_opt_ IPlatformAgnosticResource **fieldIterator)
+    {
+        UErrorCode status = U_ZERO_ERROR;
+        UDateFormat *dtf = UNWRAP_COBJECT(resource, UDateFormat);
+
+        UFieldPositionIterator *fpi = nullptr;
+
+        if (fieldIterator)
+        {
+            fpi = ufieldpositer_open(&status);
+            ICU_ASSERT(status, true);
+            IPlatformAgnosticResource *fpiResource = new IcuCObject<UFieldPositionIterator>(fpi, &ufieldpositer_close);
+            AssertOrFailFastMsg(fpiResource, "Out of memory");
+            *fieldIterator = fpiResource;
+        }
+
+        int required = udat_formatForFields(dtf, date, reinterpret_cast<UChar *>(formatted), formattedLen, fpi, &status);
+        if (formatted == nullptr && formattedLen == 0 && status == U_BUFFER_OVERFLOW_ERROR)
+        {
+            // when we are just counting bytes, we can ignore errors
+            AssertOrFailFast(required > 0);
```

I'm confused by this comment... it doesn't look like ignoring the error :P

---

we can ignore *ICU errors. oops.